utilities_test.go

```diff
@@ -98,84 +98,112 @@ func TestDatasetWarp(t *testing.T) {
+
+	// read part warp result
+	err = outputDataset.Read(0, 0, data, ds1.Structure().SizeX, ds1.Structure().SizeY,Bands(0, 1, 2))
+	assert.NoError(t, err)
+
+	for _, px := range data {
+		if px != 200 {
+			t.Errorf("wrong pixel value : %d instead of 200", px)
+		}
 	}
 
-	if geoTransform != [6]float64{45.000000, 0.250000, 0.000000, 35.000000, 0.000000, -0.250000} {
-		t.Errorf("wrong geotransform : [ %f, %f, %f, %f, %f, %f ]", geoTransform[0], geoTransform[1], geoTransform[2], geoTransform[3], geoTransform[4], geoTransform[5])
+	err = outputDataset.Read(5, 0, data, ds2.Structure().SizeX, ds2.Structure().SizeY,Bands(0, 1, 2))
+	assert.NoError(t, err)
+
+	for _, px := range data {
+		if px != 100 {
+			t.Errorf("wrong pixel value : %d instead of 100", px)
+		}
 	}
 }
 func TestDatasetWarpInto(t *testing.T) {
-	filePath := path.Join(os.TempDir(), "warpinto.tif")
-	outputDataset, err := Create(GTiff, filePath, 4, Byte, 10, 10)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	defer os.Remove(filePath)
-	defer outputDataset.Close()
+	outputDataset, _ := Create(Memory, "", 1, Byte, 5, 5)
+	inputDataset, _ := Create(Memory, "", 1, Byte, 5, 5)
 
-	sr, err := NewSpatialRefFromEPSG(4326)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = outputDataset.SetSpatialRef(sr); err != nil {
-		t.Fatal(err)
+	for _, b := range outputDataset.Bands() {
+		_ = b.Fill(200, 0)
 	}
-	if err = outputDataset.SetGeoTransform([6]float64{45.000000, 1.0, 0.000000, 35.000000, 0.000000, -1.0}); err != nil {
-		t.Fatal(err)
+
+	for _, b := range inputDataset.Bands() {
+		_ = b.Fill(155, 0)
 	}
 
-	inputDataset, _ := Open("testdata/test.tif", RasterOnly())
+	sr, _ := NewSpatialRefFromEPSG(4326)
+	_ = outputDataset.SetSpatialRef(sr)
+	_ = outputDataset.SetGeoTransform([6]float64{45, 1, 0, 35, 0, -1})
+	_ = inputDataset.SetSpatialRef(sr)
+	_ = inputDataset.SetGeoTransform([6]float64{45, 1, 0, 35, 0, -1})
+
+	defer outputDataset.Close()
 	defer inputDataset.Close()
 
 	// Warp existing dataset with multiple input dataset
-	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-co", "TILED=YES"}); err == nil {
-		t.Error("All options related to creation ignored in update mode")
-	}
+	err := outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-co", "TILED=YES"})
+	assert.Error(t, err, "All options related to creation ignored in update mode")
 
-	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-of", "GTiff"}); err == nil {
-		t.Error("All options related to creation ignored in update mode")
-	}
-
-	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-dstalpha"}, ConfigOption("GDAL_CACHEMAX=64")); err != nil {
+	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{}, ConfigOption("GDAL_CACHEMAX=64")); err != nil {
 		t.Fatal(err)
 	}
 
-	if outputDataset.Structure().NBands != 4 {
+	if outputDataset.Structure().NBands != 1 {
```

Comment:
This test can be removed completely. The number of bands of this dataset is not affected by the warpinto operation